analyses/tcd/metrics/TCD_trigger_report.Rmd

```diff
@@ -0,0 +1,432 @@
+  output <- plotTradeoffCI(trigger_id = trigger, 
+                   left_metric_name = metric)
+  
+  # save plot as png
+  filename <- paste0(trigger, "_", metric, "_tradeoff_bar.png")
+  png(filename = paste0("plots/", filename), width = 500, height = 400, units = "px")
+  print(output$p)
+  invisible(dev.off())
+  }
+}
+
+```
+
+```{r file-formatting-and-saving}
+
+# list plots to be trimmed
+plots_to_trim <- grep(list.files(path = "./plots"), 
+                      pattern = 'trimmed_', 
+                      invert = TRUE, 
+                      value = TRUE) # excludes 'trimmed_' graphs, which will be overwritten
+
+# trim plots
+for (filename in plots_to_trim) {
+
+  # read in the image
+  original_plot <- magick::image_read(paste0("plots/", filename))
+  
+  # trim the image
+  trimmed_plot <- magick::image_trim(original_plot)
+  
+  # save trimmed plot
+  magick::image_write(trimmed_plot, path = paste0("plots/trimmed_", filename), format = "png")
+
+}
+
+```
+
+```{r key-messages-metrics}
+t3_atv_low_end_68 <- perf_metrics_data %>% filter(trigger == 'Trigger3' & metric == 'atv') %>% filter(upoint == 'low_end_68') %>% select(value) %>% as.numeric()
+
+t3_atv_high_end_68 <- perf_metrics_data %>% filter(trigger == 'Trigger3' & metric == 'atv') %>% filter(upoint == 'high_end_68') %>% select(value) %>% as.numeric()
+
+t12_atv_central <- perf_metrics_data %>% filter(trigger == 'Trigger1-2' & metric == 'atv') %>% filter(upoint == 'central_95') %>% select(value) %>% as.numeric()
+t3_det_low_end_68 <- perf_metrics_data %>% filter(trigger == 'Trigger3' & metric == 'det') %>% filter(upoint == 'low_end_68') %>% select(value) %>% as.numeric()
+
+t3_det_high_end_68 <- perf_metrics_data %>% filter(trigger == 'Trigger3' & metric == 'det') %>% filter(upoint == 'high_end_68') %>% select(value) %>% as.numeric()
+
+t3_far_low_end_68 <- perf_metrics_data %>% filter(trigger == 'Trigger3' & metric == 'far') %>% filter(upoint == 'low_end_68') %>% select(value) %>% as.numeric()
+
+t3_far_high_end_68 <- perf_metrics_data %>% filter(trigger == 'Trigger3' & metric == 'far') %>% filter(upoint == 'high_end_68') %>% select(value) %>% as.numeric()
+
+mitigation_warranted <- ifelse(t3_far_low_end_68 >= 40, "Mitigation is recommended to minimise the negative impact of a false alarm due to its moderate to high likelihood.", "")
+
+```
+
+# Anticipatory Action Trigger Mechanism Report
+<p style="color:#007ce1;font-size:22px;text-align:center;"> `r paste0(shock, ' in ', country_name)`</p> 
+<p style="color:black;font-size:14px;text-align:right;"> Last updated: `r Sys.Date()`</p> 
+
+## Key Messages
+* <span style="color: blue;">LIMITED HISTORICAL DATA</span> <p style="color:black;font-size:16px;text-align:left;"> Triggers 1 and 2 rely on the IRI seasonal forecast, with limited data from 2018 to today. Likelihood of activation estimated based on global data, for Triggers 1 and 2 together, but not together with Trigger 3, and no performance metrics are calculated.</p>
```

Comment:
I thought historical forecasts went back to 2017?

Reply:
Unpack the second sentence. What is "global" referring to? WHy split T1+T2 from T3? Why no performance metrics? What does it mean for our understanding of likelihood and risk in disbursing funds based on this mechanism? RIght now it reads more like a footnote than an important warning on limitations.

Also this is the key messages section. Consider breaking it up the first point and/or develop to provide more context later in the doc.

Reply:
Historical forecasts didn't include the periods for 2017 that we are interested in, so for this pilot, data was only available until 2017.

Have made this read more like a key point and will add more context down below in the document.